step-plans/step-plans-core/src/main/java/step/core/plans/runner/PlanRunnerResult.java

```diff
@@ -72,15 +76,42 @@ public PlanRunnerResult(String executionId, String rootReportNodeId, ReportTreeA
 	}
 	
 	public ReportNodeStatus getResult() {
-		ReportNode rootReportNode = reportTreeAccessor.get(rootReportNodeId);
+		ReportNode rootReportNode = getRootReportNode();
 		if(rootReportNode != null) {
-			ReportNodeStatus status = rootReportNode.getStatus();
-			return status;
+			return rootReportNode.getStatus();
 		} else {
 			return ReportNodeStatus.NORUN;
 		}
 	}
 
+	public ReportNode getRootReportNode() {
+		return reportTreeAccessor.get(rootReportNodeId);
+	}
+
+	/**
+	 * @return a stream of all report nodes with errors that caused this execution to fail.
+	 * This returns only the so-called contributing errors according to following definition {@link ReportNode#getContributingError()}
+	 */
+	public Stream<ReportNode> getReportNodesWithErrors() {
+		return getReportTreeAccessor().getReportNodesWithContributingErrors(executionId);
+	}
+
+	/**
+	 * @return a stream of all errors that caused this execution to fail.
+	 * This returns only the so-called contributing errors according to following definition {@link ReportNode#getContributingError()}
+	 */
+	public Stream<Error> getErrors() {
+		return getReportNodesWithErrors().map(ReportNode::getError).filter(Objects::nonNull);
+	}
+
+	/**
+	 * @return a concatenation of the first 10 error messages that caused this execution to fail.
+	 * This considers only the so-called contributing errors according to following definition {@link ReportNode#getContributingError()}
+	 */
+	public String getErrorSummary() {

```

Comment:
I assume we don't have yet a concept of severity to order by severity before limiting to 10, right?